pkg/kubectl/resource/builder.go

```diff
@@ -274,6 +276,31 @@ func (b *Builder) LabelSelector(selector labels.Selector) *Builder {
 	return b
 }
 
+// FieldSelectorParam defines a selector that should be applied to the object types to load.
+// This will not affect files loaded from disk or URL. If the parameter is empty it is
+// a no-op - to select all resources invoke `b.Selector(fields.Everything)`.
+func (b *Builder) FieldSelectorParam(s string) *Builder {
+	selector, err := fields.ParseSelector(s)
```

Comment:
We should not be parsing field selectors on the client side - instead, field selectors should be passed to the server unmodified.  If we can't pass an unmodified field selector to the server without parsing it, then we need to pick the minimum possible constraints on the client side.

Reply:
This follows current `SelectorParam` for `labels`.

https://github.com/kubernetes/kubernetes/blob/e821e531db24ddd4897a166b67af8c863d93edc2/pkg/kubectl/resource/builder.go#L252-L269

> We should not be parsing field selectors on the client side - instead, field selectors should be passed to the server unmodified.

@smarterclayton Also apply to `labelSelector`?

Reply:
Yes that should probably apply to label selectors as well.  The server is responsible for handling label selection for the most part.  There is some level of client side processing possible, but for the most part label selector should be able to be passed opaquely.